File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    queue = deque([(p, i) for i, p in enumerate(priorities)])
    answer = 0

    while queue:
        current = queue.popleft()
        if any(current[0] < q[0] for q in queue):  # 우선순위 높은 게 있으면 뒤로 보냄
            queue.append(current)
        else:
            answer += 1
            if current[1] == location:  # 우리가 찾는 인덱스면 종료
                return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let queue = priorities.map((priority, index) => [priority, index]);
    let answer = 0;

    while (queue.length > 0) {
        let [currentPriority, currentIndex] = queue.shift();
        
        // 뒤에 더 높은 우선순위가 있는지 확인
        if (queue.some(([priority]) => priority > currentPriority)) {
            queue.push([currentPriority, currentIndex]); // 뒤로 보냄
        } else {
            answer += 1; // 실행 처리
            if (currentIndex === location) {
                return answer;
            }
        }
    }
}
